cumulus/client/cirrus-executor/src/aux_schema.rs

```diff
@@ -208,6 +208,92 @@ pub(super) fn delete_bad_receipt<Backend: AuxStore>(
 	)
 }
 
+fn delete_expired_bad_receipt_info_at<Backend: AuxStore, Number: Encode>(
+	backend: &Backend,
+	block_number: Number,
+) -> Result<(), sp_blockchain::Error> {
+	let block_number_key = (BAD_RECEIPT_BLOCK_NUMBER, block_number).encode();
+
+	let bad_receipt_hashes: Vec<H256> =
+		load_decode(backend, block_number_key.as_slice())?.unwrap_or_default();
+
+	let keys_to_delete = bad_receipt_hashes
+		.into_iter()
+		.map(bad_receipt_mismatch_index_key)
+		.chain(std::iter::once(block_number_key))
+		.collect::<Vec<_>>();
+
+	backend.insert_aux([], &keys_to_delete.iter().map(|k| &k[..]).collect::<Vec<_>>()[..])
+}
+
+/// Returns the first unconfirmed bad receipt info necessary for building a fraud proof if any.
+///
+/// Bad receipts which are older than `oldest_receipt_number` are expired and will be pruned.
```

Comment:
Why is this pruning necessary inside of `load` method?
It feels like two independent things conflated into one for reason I don't understand.

Reply:
Okay, looks fine to me, then we'll first process the potentially expired bad receipts and load the first unconfirmed one after that.

Reply:
https://github.com/subspace/subspace/pull/652/commits/e60d5b4c780a8fbe9ad6310cdf8f6b9f1f01d43b